packages/db-migrations/databases/fxa/patches/patch-164-165.sql

```diff
@@ -0,0 +1,101 @@
+SET NAMES utf8mb4 COLLATE utf8mb4_bin;
+
+CALL assertPatchLevel('164');
+
+-- Add verificationMethod to password forgot token table and account reset token table
+ALTER TABLE `passwordForgotTokens` ADD COLUMN `verificationMethod` INT DEFAULT NULL, ALGORITHM = INPLACE, LOCK = NONE;
+ALTER TABLE `accountResetTokens` ADD COLUMN `verificationMethod` INT DEFAULT NULL, ALGORITHM = INPLACE, LOCK = NONE;
+
+-- Update passwordForgotToken procedure to return verificationMethod
+CREATE PROCEDURE `passwordForgotToken_3` (
+    IN `inTokenId` BINARY(32)
+)
+BEGIN
+SELECT
+    t.uid,
+    t.tokenData,
+    t.createdAt,
+    t.passCode,
+    t.tries,
+    e.email,
+    a.verifierSetAt,
+    t.verificationMethod
+FROM
+    passwordForgotTokens t,
+    accounts a,
+    emails e
+WHERE
+    t.tokenId = inTokenId
+  AND
+    t.uid = a.uid
+  AND
+    t.uid = e.uid
+  AND
+    e.isPrimary = true
+;
+END;
+
+-- Update accountResetToken procedure to return verificationMethod
+CREATE PROCEDURE `accountResetToken_2` (
+    IN `inTokenId` BINARY(32)
+)
+BEGIN
+SELECT
+    t.uid,
+    t.tokenData,
+    t.createdAt,
+    a.verifierSetAt,
+    t.verificationMethod
+FROM
+    accountResetTokens t,
+    accounts a
+WHERE
+    t.tokenId = inTokenId
+  AND
+    t.uid = a.uid
+;
+END;
+
+CREATE PROCEDURE `forgotPasswordVerified_9` (
+    IN `inPasswordForgotTokenId` BINARY(32),
+    IN `inAccountResetTokenId` BINARY(32),
+    IN `inTokenData` BINARY(32),
+    IN `inUid` BINARY(16),
+    IN `inCreatedAt` BIGINT UNSIGNED,
+    IN `inVerificationMethod` INT
+        )
+BEGIN
+    DECLARE EXIT HANDLER FOR SQLEXCEPTION
+BEGIN
+        -- ERROR
+ROLLBACK;
+RESIGNAL;
+END;
+
+START TRANSACTION;
+
+-- Pass `inVerificationMethod` to the password forgot token table
+REPLACE INTO accountResetTokens(
```

Comment:
To reduce complexity and token management, we can copy the verification state of the passwordForgotToken into the accountResetToken.